docs/designers-developers/developers/themes/block-based-themes.md

```diff
@@ -0,0 +1,96 @@
+
+> Documentation has been shared early to surface what’s being worked on and invite feedback from those experimenting with the APIs. You can provide feedback in the weekly #core-editor chats where the latest progress of this effort will be shared and discussed, or async via Github issues.
+
+## What is a block-based theme?
+
+A block-based theme is a WordPress theme with templates entirely composed of blocks so that in addition to the post content of the different post types (pages, posts, ...), the block editor can also be used to edit all areas of the site: headers, footers, sidebars, etc.
+
+## What is the structure of a block-based theme?
+
+A very simple block-based theme is structured like so:
+
+```
+theme
+|__ style.css
+|__ functions.php
+|__ block-templates
+    |__ index.html
+    |__ single.html
+    |__ archive.html
+    |__ ...
+|__ block-template-parts
+    |__ header.html
+    |__ footer.html
+    |__ sidebar.html
+    |__ ...
+```
+
+The difference with existing WordPress themes is that the different templates in the template hierarchy, and template parts, are block templates instead of php files.
+
+## What is a block template?
+
+A block template is made up of a list of blocks. Any WordPress block can be used in a template. Templates can also reuse parts of their content using "Template Parts". For example, all the block templates can have the same header included from a separate `header.html` template part.
+
+Here's an example of a block template:
+
+```html
+<!-- wp:site-title /-->
+
+<!-- wp:image {"sizeSlug":"large"} -->
+<figure class="wp-block-image size-large">
+	<img src="https://cldup.com/0BNcqkoMdq.jpg" alt="" />
+</figure>
+<!-- /wp:image -->
+
+<!-- wp:group -->
+<div class="wp-block-group">
+	<div class="wp-block-group__inner-container">
+		<!-- wp:post-title /-->
+		<!-- wp:post-content /-->
+	</div>
+</div>
+<!-- /wp:group -->
+
+<!-- wp:group -->
+<div class="wp-block-group">
+	<div class="wp-block-group__inner-container">
+		<!-- wp:heading -->
+		<h2>Footer</h2>
+		<!-- /wp:heading -->
+	</div>
+</div>
+<!-- /wp:group -->
+```
+
+## How to write and edit these templates?
+
+Ultimately, any WordPress user with the correct capabilities (example: `administrator` WordPress role) will be able to access these templates in the WordPress admin, edit them in dedicated views and potentially export them as a theme.
+
+In the current iteration (at the time of writing this doc), you can navigate to the temporary "Templates" admin menu under "Appearance" `wp-admin/edit.php?post_type=wp_template` and use this as a playground to edit your templates.
```

Comment:
Just noting that the "Full Site Editing" feature flag must be enabled in the _Gutenberg > Experiments_ settings for the temporary _Templates_ menu to appear.

Reply:
That should be mentioned in the introduction. None of this will work without enabling it.